spicy/toolchain/src/ast/types/unit.cc

```diff
@@ -38,6 +40,27 @@ hilti::optional_ref<const type::unit::Item> type::Unit::itemByName(const ID& id)
     return {};
 }
 
+std::optional<uint64_t> type::Unit::size() const {
+    uint64_t result = 0;
+    for ( const auto& i : this->itemRefs() ) {
+        if ( const auto& f = i->tryAs<unit::item::Field>() ) {
+            // TODO(bbannier): To support fields with conditions we need to be able to return an runtime value (an
+            // `Expression`) instead of a compile-time `uint64_t` here which requires support in the type API.
```

Comment:
This seems tricky, not sure if it's even viable at all. Maybe just skip the comment and focus this just on static properties.